web/github.go

```diff
@@ -70,45 +131,153 @@ func (g *GitHub) Orgs(w http.ResponseWriter, r *http.Request) {
+				ID:   installation.GetAccount().GetID(),
+				Name: installation.GetAccount().GetLogin(),
+			})
 		}
 	}
 
 	successJSON(w, r, orgs)
 }
+
+// TODO (@carlosms) check if this makes more sense as a middleware
+// orgInstallation returns the GitHub App Installation corresponding to the URL
+// parameter "orgName". The logged-in user must be an administrator of the
+// organization. If there is any error the proper HTTP headers are set in w.
+func (g *GitHub) orgInstallation(w http.ResponseWriter, r *http.Request) (*github.Installation, error) {
+	login, err := GetUserLogin(r.Context())
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed to get user login from context")
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return nil, fmt.Errorf(http.StatusText(http.StatusUnauthorized))
+	}
+
+	orgName := chi.URLParam(r, "orgName")
+	if orgName == "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return nil, fmt.Errorf(http.StatusText(http.StatusBadRequest))
+	}
+
+	installation, err := g.installation(r.Context(), orgName)
+	if err != nil {
+		if err == errNotFound {
+			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+			return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+		}
+
+		ctxlog.Get(r.Context()).Errorf(err, "failed to get installation with name %v", orgName)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return nil, fmt.Errorf(http.StatusText(http.StatusInternalServerError))
+	}
+
+	// a GitHub App can also be installed in a User account
+	if installation.GetAccount().GetType() != "Organization" {
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+	}
+
+	admin, err := g.admin(w, r, installation, login)
+	if err != nil {
+		ctxlog.Get(r.Context()).Errorf(err, "failed to check user admin role")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return nil, fmt.Errorf(http.StatusText(http.StatusInternalServerError))
+	}
+
+	if !admin {
+		// 404 to avoid leaking to a non admin if the org exists or is installed
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return nil, fmt.Errorf(http.StatusText(http.StatusNotFound))
+	}
+
+	return installation, nil
+}
+
+// orgResponse is the response type used by the individual organization handler
+type orgResponse struct {
+	ID     int64  `json:"id"`
+	Name   string `json:"name"`
+	Config string `json:"config"`
+}
+
+// Org writes in the response the individual organization requested by the URL
+// parameter "orgName", only if the user is an admin
+func (g *GitHub) Org(w http.ResponseWriter, r *http.Request) {
+	installation, err := g.orgInstallation(w, r)
+	if err != nil {
+		return
+	}
+
+	idStr := strconv.FormatInt(installation.GetAccount().GetID(), 10)
```

Comment:
it's a string, isn't it? https://github.com/src-d/lookout/pull/483/files#diff-d87db3c51b2eaa55b7443de6a8e0f17cR33